src/main/java/baritone/command/defaults/FarmCommand.java

```diff
@@ -34,8 +43,50 @@ public FarmCommand(IBaritone baritone) {
 
     @Override
     public void execute(String label, IArgConsumer args) throws CommandException {
-        args.requireMax(0);
-        baritone.getFarmProcess().farm();
+        args.requireMax(2);
+        int range = 0;
+        BetterBlockPos origin;
+        //range
+        if (args.has(1)) {
+            range = args.getAs(Integer.class);
+        }
+        //waypoint
+        if (args.hasExactly(1)){
+            IWaypoint[] waypoints = args.getDatatypeFor(ForWaypoints.INSTANCE);
+            IWaypoint waypoint = null;
+            if (args.hasAny() && args.peekString().equals("@")) {
```

Comment:
Is this pasted from somewhere? Should it be a datatype (i don't really know the furry command system that well)?

Just feels like this isn't dry

Reply:
The waypoints implementation is based on `WaypointsCommand.java` line `152` to `177`.
`args.hasExactly(1)` can be replaced by `args.has(1)` if it's that you're referring to.

Reply:
It was the timestamp compare system, have removed it now.